### PR TITLE
Add CI for examples

### PR DESCRIPTION
**Pull request by djacu**

### CODE REVIEW

File: lib.nix [hunk 1]
```diff
@@ -77,7 +77,7 @@ let
           in
 
           let nickel_expr | params.nix.NickelExpression =
-            import "${sources}/${nickelFile}"
+            import "${nickelFile}"
```
Comment (rapenne-s) [inline lib.nix:80]: fixing ` > error: import of /nix/store/p6qc3j8vgfj1c918m04mg4hgsln4lwq0-c-hello-world//nix/store/i4bm1rmvjn37iz4fmn7bshwb3jgfg81x-hello.ncl failed: No such file or directory (os error 2)`

File: lib.nix [hunk 2]
```diff
@@ -96,7 +96,7 @@ let
       };
       fileToCall = builtins.toFile "extract-inputs.ncl" ''
         let contracts = import "${./.}/contracts.ncl" in
-        let nickel_expr = import "${sources}/${nickelFile}" in
+        let nickel_expr = import "${nickelFile}" in
```
Comment (rapenne-s) [inline lib.nix:99]: fixing `fixing ` > error: import of /nix/store/p6qc3j8vgfj1c918m04mg4hgsln4lwq0-c-hello-world//nix/store/i4bm1rmvjn37iz4fmn7bshwb3jgfg81x-hello.ncl failed: No such file or directory (os error 2)``

File: lib.nix [hunk 3]
```diff
@@ -130,7 +130,7 @@ let
           # baseDir explicitly? Maybe, but the issue is that this path is the
           # path of the git directory, not the subdirectory of the flake.nix.
           # may need some massaging
-          let as_nix_path = baseDir + "/${declaredInputs.${inputId}.path}";
+          let as_nix_path = baseDir + "/${builtins.concatStringsSep "." declaredInputs.${inputId}.path}";
```
Comment (rapenne-s) [inline lib.nix:133]: fixing `error: cannot coerce a list to a string`

**Comment by yannham** [comment]
Yes, we changed quite a bit of things. For once, `output` is not a function anymore. I'm curious though, did you print the full build logs? Was it understandable? I'm interested to see if error reporting is helpful in practice

**Review by yannham — APPROVED** [review]

**Pull request merged**